tests/test_clarity.py

```diff
@@ -253,11 +253,6 @@ def test_get_samples_sequenced_with(self, mocked_get_sample, mocked_containers,
         mocked_names_from_plate.assert_any_call('this')
         mocked_names_from_plate.assert_any_call('that')
 
-    @patched_lims('get_processes', [FakeProcess])
-    def test_get_released_samples(self, mocked_lims):
-        assert clarity.get_released_samples() == ['that', 'this']
-        mocked_lims.assert_called_with(type='Data Release EG 1.0')
-
     @patched_clarity('get_sample', Mock(artifact=Mock(id='an_artifact_id')))
     @patched_lims('get_processes', side_effect=[[FakeProcess], [FakeProcess, FakeProcess(date_run='another_date')]])
```

Comment:
get_sample_release_date now does two calls to get_processes, so the side_effect here gets exhausted early. The following fixes it:

```python
@patched_lims('get_processes', side_effect=[[FakeProcess], [], [], [FakeProcess, FakeProcess(date_run='another_date')]])
def test_get_sample_release_date(self, mocked_get_procs, mocked_get_sample):
    assert clarity.get_sample_release_date('a_sample_name') == 'a_date'
    for step_name in ['Data Release EG 1.0', 'Data Release EG 2.0']:
        mocked_get_procs.assert_any_call(type=step_name, inputartifactlimsid='an_artifact_id')
...
```

Reply:
Thanks
That should be done now